lib/AST/ProtocolConformance.cpp

```diff
@@ -288,6 +286,45 @@ GenericSignature *ProtocolConformance::getGenericSignature() const {
   llvm_unreachable("Unhandled ProtocolConformanceKind in switch.");
 }
 
+SubstitutionMap ProtocolConformance::getSubstitutions(ModuleDecl *M) const {
```

Comment:
There's an existing `SpecializedProtocolConformance::getSubstitutionMap()`. Should that be generalized to cover this case as well, i.e., by pushing it up to `ProtocolConformance::getSubstitutionMap()`?

Reply:
`SpecializedProtocolConformance::getSubstitutionMap` doesn't need the `ModuleDecl`, so I think it makes sense to retain it at least as an implementation detail of this and for internal use in `SpecializedProtocolConformance`.